client-common/src/main/java/org/apache/cassandra/sidecar/common/request/data/CreateRestoreJobRequestPayload.java

```diff
@@ -205,31 +226,48 @@ public static class Builder
 
         public Builder jobId(UUID jobId)
         {
-            this.jobId = jobId;
-            return this;
+            return update(b -> b.jobId = jobId);
         }
 
         public Builder jobAgent(String jobAgent)
         {
-            this.jobAgent = jobAgent;
-            return this;
+            return update(b -> b.jobAgent = jobAgent);
         }
 
         public Builder updateImportOptions(Consumer<SSTableImportOptions> updater)
         {
-            updater.accept(importOptions);
-            return this;
+            return update(b -> updater.accept(b.importOptions));
         }
 
-        public Builder consistencyLevel(String consistencyLevel)
+        public Builder consistencyLevel(ConsistencyLevel consistencyLevel)
+        {
+            return consistencyLevel(consistencyLevel, null);
+        }
+
+        public Builder consistencyLevel(ConsistencyLevel consistencyLevel, String localDc)
+        {
+            return update(b -> {
+                b.consistencyLevel = consistencyLevel;
+                b.localDc = localDc;
+            });
+        }
+
+        @Override
+        public Builder self()
         {
-            this.consistencyLevel = consistencyLevel;
             return this;
         }
 
         public CreateRestoreJobRequestPayload build()
         {
+            Preconditions.checkArgument(!consistencyLevel.isLocalDcOnly || (localDc != null && !localDc.isEmpty()),
+                                        "Must specify a non-empty localDc for consistency level: " + consistencyLevel);
```

Comment:
Nit: `"Must specify a non-empty " + JOB_LOCAL_DATA_CENTER + " for consistency level: "`?